8/src/main/java/com/traveltime/plugin/solr/query/TravelTimeScorer.java

```diff
@@ -0,0 +1,255 @@
+
+        return docValues.advanceExact(docID()) ? score(travelTime) : 0;
+    }
+
+    @Override
+    public DocIdSetIterator iterator() {
+        // add indirection so that if 'it' is updated then it will
+        // be taken into account
+        return new DocIdSetIterator() {
+
+            @Override
+            public int nextDoc() throws IOException {
+                return doc = it.nextDoc();
+            }
+
+            @Override
+            public int docID() {
+                return doc;
+            }
+
+            @Override
+            public long cost() {
+                return it.cost();
+            }
+
+            @Override
+            public int advance(int target) throws IOException {
+                return doc = it.advance(target);
+            }
+        };
+    }
+
+    @Override
+    public float getMaxScore(int upTo) {
+        return boost;
+    }
+
+    private int setMinCompetitiveScoreCounter = 0;
+
+
+    @Override
+    public void setMinCompetitiveScore(float minScore) throws IOException {
+        if (minScore > boost) {
+            it = DocIdSetIterator.empty();
+            return;
+        }
+
+        setMinCompetitiveScoreCounter++;
+        // We sample the calls to this method as it is expensive to recalculate the iterator.
+        if (setMinCompetitiveScoreCounter > 256 && (setMinCompetitiveScoreCounter & 0x1f) != 0x1f) {
+            return;
+        }
+
+        int previousMaxTravelTime = maxTravelTime;
+        maxTravelTime = computeMaxTravelTime(minScore, maxTravelTime);
+        if (maxTravelTime == previousMaxTravelTime) {
+            // nothing to update
+            return;
+        }
+
+        DocIdSetBuilder result = new DocIdSetBuilder(maxDoc);
+        final int doc = docID();
+        PointValues.IntersectVisitor visitor = new PointValues.IntersectVisitor() {
+
+            DocIdSetBuilder.BulkAdder adder;
+
+            @Override
+            public void grow(int count) {
+                adder = result.grow(count);
+            }
+
+            @Override
+            public void visit(int docID) {
+                if (docID <= doc) {
+                    // Already visited or skipped
+                    return;
+                }
+                adder.add(docID);
+            }
+
+            @Override
+            public void visit(int docID, byte[] packedValue) {
+                if (docID <= doc) {
+                    // Already visited or skipped
+                    return;
+                }
+
+                int documentTravelTime = getTravelTimeForCoordinate(packedValue);
+
+                if (documentTravelTime == -1 || documentTravelTime > maxTravelTime) {
+                    //Document travel time is longer than maximum travel time
+                    return;
+                }
+                adder.add(docID);
+            }
+
+            @Override
+            public PointValues.Relation compare(byte[] rangeMinPackedValue, byte[] rangeMaxPackedValue) {
```

Comment:
@jstuyts I'm not sure if I understand what this intersection test should mean.
If this checks whether the query results intersect with a rectangle given by the coordinate pair `rangeMinPackedValue`, `rangeMaxPackedValue`? In that case I think the shapes generated by a search (especially for public transport) can be complex enough that we should always return `CELL_CROSSES_QUERY`